### PR TITLE
fix(table):fix issue with abnormal string output of 'lv_table_set_cell_value_fmt'

### PR DESCRIPTION
**Pull request by lhdjply**
fix(table):fix issue with abnormal string output of 'lv_table_set_cell_value_fmt'

**test code：**
```c
void lv_table_test(void)
{
    lv_obj_t *table = lv_table_create(lv_screen_active());
    lv_obj_set_style_pad_all(table, 8, LV_PART_ITEMS);
    lv_obj_set_style_text_align(table, LV_TEXT_ALIGN_CENTER, LV_PART_ITEMS);
    lv_obj_set_size(table, 280, 280);
    lv_obj_center(table);

    for(int i;i<20;i++)
    {
        lv_table_set_cell_value_fmt(table,
                            i,
                            0,
                            "%d",
                            i);

        lv_table_set_cell_value_fmt(table,
                            i,
                            1,
                            "%.3f",
                            0.123f+i);
    }
}
```
**Picture before fix：**
屏幕截图 2023-11-11 091925

**Picture after fix：**
屏幕截图 2023-11-11 091509

Similarly, the **V8 version** also has the same issue

### CODE REVIEW

**Comment by lhdjply** [comment]
@kisvegabor Is there any problem with this PR?

File: src/widgets/table/lv_table.c [hunk 1]
```diff
@@ -178,16 +178,16 @@ void lv_table_set_cell_value_fmt(lv_obj_t * obj, uint32_t row, uint32_t col, con
...
     if(table->cell_data[cell] == NULL) {
         va_end(ap2);
         return;
     }
 
-    table->cell_data[cell]->txt[len] = 0; /*Ensure NULL termination*/
+    table->cell_data[cell]->txt[len + 1] = 0; /*Ensure NULL termination*/
 
-    lv_vsnprintf(table->cell_data[cell]->txt, len, fmt, ap2);
+    lv_vsnprintf(table->cell_data[cell]->txt, len + 1, fmt, ap2);
```
Comment (kisvegabor) [inline src/widgets/table/lv_table.c:190]: I've tried out the test code and it seems to me we need to modify only this, to tell that we have space for the `\0` as well.

**Review by kisvegabor — COMMENTED** [review]
Hi,

Thanks for pinging me, somehow I missed this PR.

File: src/widgets/table/lv_table.c (hunk 1 above)
Comment (lhdjply) [reply src/widgets/table/lv_table.c:190]: Oh yes,I'll make some modifications

**Comment by kisvegabor** [comment]
Thank you! ^^

**Comment by lhdjply** [comment]
@kisvegabor  I think the issue with V8.3 should also be fixed together

**Comment by kisvegabor** [comment]
Oh, yes, could you send a PR there as well?

**Comment by lhdjply** [comment]
> Oh, yes, could you send a PR there as well?

OK

**Pull request merged**